docs/guide/writing-tests-for-modules.md

```diff
@@ -0,0 +1,236 @@
+# Writing tests for modules
+
+Tests are useful to guarantee that the requirements of your application are met over time.
+
+You don't have to test over and over again the same scenarios, it's a time saving tool.
+
+There are many kind of tests
+
+- [unit testing](https://en.wikipedia.org/wiki/Unit_testing) where we test a single part of the application, a single
+    module, file commonly referred as a unit
+- [integration testing](https://en.wikipedia.org/wiki/Integration_testing) where we test multiple parts of an
+    application working together as a group
+- [system testing](https://en.wikipedia.org/wiki/System_testing), sometimes referred as [end-to-end testing](https://www.indeed.com/career-advice/career-development/end-to-end-testing) where we test the entire application
+- and many more
+
+This documentation focuses on how to test Apostrophe modules, not Apostrophe itself using integration testing.
+
+Modules should be stand-alone npm packages.
```

Comment:
Can we expand this to something like: "Modules should be stand-alone npm packages. They need to be hosted somewhere that our testing can access. For this example, we are hosting this module in a github repo. We will use a fictitious module named `article` to illustrate how to write tests for modules."

Reply:
My bad -> "github" -> "GitHub"